### DISCUSSION TITLE
Weekly or monthly budget/hour constraints on projects

### DISCUSSION THREAD

**Discussion by pettersh** [Puzzlepart/did#1110] [Feature request]
As a delivery manager / account manager I need a way to ensure that we don't exceed the budget agreed with the customer.
This is easy when I'm the only one registering hours on a project, but it's harder when we are more than one.

I wish we had a way to set a weekly or hourly budget on a project and some system support to help me ensure we don't burn the budget to fast. Ideas could be:

- dashboard or graphs showing actual burndown and trends, like in Azure
- notifications triggered when burnrate is too high, ideally in Teams, but email is ok, a power platform connector could also be a good idea
- making the project not available for time registering if exceeded? Maybe a warning before submitting the timesheet

Example/inspiration:

![image](https://github.com/Puzzlepart/did/assets/2064929/88378e2e-315b-4684-8712-23f9b8ce62e2)

Here's a short video about budgeting in Azure. I don't need anything this advanced, but wanted to show one way of doing this
https://www.youtube.com/watch?v=UrkHiUx19Po

**Comment by pettersh** [comment]
fyi @okms

**Comment by olemp** [comment]
Does it need to be per week or month or we could we start with total budget for the project? Perhaps this is out of scope for did, and should be part of Dynamics @okms @damsleth?

**Reply by pettersh** [reply]
Dynamics doesn't have any data about the hours we actually register, so did is the only place where we have this kind of data.
Per month notifications doesn't give me the control I need to stay within the budget.

I'm not sure what would be the easiest route to success here, but maybe a weekly email notification if hours registered last week exceeds the threshold/budget. If som kind of updated monthly forecast based on actual hours current month is easier, then that would be good too.

Maybe this functionality could be implemented using logic apps or power automate, but the did data model would need to support the scenario. A **did connector for power platform** would be awesome, because working with GraphQL is too difficult for non-tecnical people, and even technical people like myself.
